apps/emqx_opentelemetry/src/emqx_otel_schema.erl

```diff
@@ -42,41 +74,127 @@ fields("opentelemetry") ->
+                    default => false,
+                    desc => ?DESC(enable),
+                    importance => ?IMPORTANCE_HIGH
+                }
+            )},
+        {max_queue_size,
+            ?HOCON(
+                pos_integer(),
+                #{
+                    default => 2048,
+                    desc => ?DESC(max_queue_size),
                     importance => ?IMPORTANCE_HIDDEN
                 }
             )},
-        {"endpoint",
+        {exporting_timeout,
             ?HOCON(
-                emqx_schema:url(),
+                emqx_schema:timeout_duration_ms(),
                 #{
-                    mapping => "opentelemetry_exporter.otlp_endpoint",
-                    default => <<"http://localhost:4317">>,
-                    desc => ?DESC(endpoint)
+                    default => <<"30s">>,
+                    desc => ?DESC(exporting_timeout),
+                    importance => ?IMPORTANCE_HIDDEN
                 }
             )},
-        {"interval",
+        {scheduled_delay,
             ?HOCON(
                 emqx_schema:timeout_duration_ms(),
                 #{
-                    default => <<"10s">>,
-                    required => true,
-                    desc => ?DESC(interval)
+                    default => <<"1s">>,
+                    desc => ?DESC(scheduled_delay),
+                    importance => ?IMPORTANCE_HIDDEN
+                }
+            )},
+        {exporter,
+            ?HOCON(
+                ?R_REF("otel_logs_exporter"),
+                #{
+                    desc => ?DESC(exporter),
+                    importance => ?IMPORTANCE_HIGH
                 }
             )}
-    ].
+    ];
+fields("otel_metrics_exporter") ->
+    exporter_fields(metrics);
+fields("otel_logs_exporter") ->
+    exporter_fields(logs).
 
 desc("opentelemetry") -> ?DESC(opentelemetry);
 desc("exporter") -> ?DESC(exporter);
+desc("otel_logs_exporter") -> ?DESC(exporter);
+desc("otel_metrics_exporter") -> ?DESC(exporter);
+desc("otel_logs") -> ?DESC(otel_logs);
+desc("otel_metrics") -> ?DESC(otel_metrics);
 desc(_) -> undefined.
+
+exporter_fields(OtelSignal) ->
+    [
+        {endpoint,
+            ?HOCON(
+                emqx_schema:url(),
+                #{
+                    default => "http://localhost:4317",
+                    desc => ?DESC(exporter_endpoint),
+                    importance => ?IMPORTANCE_HIGH
+                }
+            )},
+        {protocol,
+            ?HOCON(
+                %% http protobuf/json may be added in future
+                ?ENUM([grpc]),
+                #{
+                    default => grpc,
+                    desc => ?DESC(exporter_protocol),
+                    importance => ?IMPORTANCE_HIDDEN
+                }
+            )},
+        {ssl_options,
+            ?HOCON(
+                ?R_REF(emqx_schema, "ssl_client_opts"),
+                #{
+                    default => #{<<"enable">> => false},
```

Comment:
it seems ssl enable/disable is derived from the endpoint url scheme.
maybe create a new type here and call emqx_schema:common_ssl_opts_schema (export it) for the fields.

i.e. no need for `enable` field.

Reply:
Fixed slightly differently, following similar cases in other schemas, e.g.: https://github.com/emqx/emqx/blob/master/apps/emqx_ldap/src/emqx_ldap.erl#L100